### PR TITLE
Document Triton dependency for the release process

### PR DESCRIPTION
**Pull request by atalman**
Document triton dependency for the Pytorch Release

### <samp>🤖 Generated by Copilot at e9773f0</samp>

Add documentation for Triton dependency in `RELEASE.md`. The documentation covers how to install and use Triton for various PyTorch builds and platforms.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -352,3 +353,19 @@ git config --file=.gitmodules -e
...
+
+In Nightly builds we have following configuration:
+* Conda builds, depend on: https://anaconda.org/pytorch-nightly/torchtriton
+* Wheel builds, depend on : https://download.pytorch.org/whl/nightly/pytorch-triton/
+* Rocm wheel builds, depend on : https://download.pytorch.org/whl/nightly/pytorch-triton-rocm/
+
+However for release we have following :
+* Conda builds, depend on: https://anaconda.org/pytorch-test/torchtriton for test and https://anaconda.org/pytorch/torchtriton for release
+* Wheel builds, depend only triton pypi package: https://pypi.org/project/triton/ for both test and release
+* Rocm wheel builds, depend on : https://download.pytorch.org/whl/test/pytorch-triton-rocm/ for test and https://download.pytorch.org/whl/pytorch-triton-rocm/ for release
```
Comment (albanD) [inline RELEASE.md:369]: Isn't this the one in PyPi?

File: RELEASE.md [hunk 2]
```diff
@@ -352,3 +353,19 @@ git config --file=.gitmodules -e
...
+
+Important: The release of https://pypi.org/project/triton/ needs to be requested from OpenAI once branch cut is completed. Please include the release PIN hash in the request: https://github.com/pytorch/pytorch/blob/release/2.1/.ci/docker/ci_commit_pins/triton.txt .
```
Comment (malfet) [inline RELEASE.md:371]: Should trition.txt point to `release/2.X` in release branch?

File: RELEASE.md [hunk 3]
```diff
@@ -352,3 +353,19 @@ git config --file=.gitmodules -e
 An example of this process can be found here:
 
 * https://github.com/pytorch/pytorch/pull/48312
+
+## Triton dependency for the release
+
+In nightly builds for conda and wheels pytorch depend on Triton build by this workflow: https://hud.pytorch.org/hud/pytorch/pytorch/nightly/1?per_page=50&name_filter=Build%20Triton%20Wheel. The pinned version of triton used by this workflow is specified here:  https://github.com/pytorch/pytorch/blob/main/.ci/docker/ci_commit_pins/triton.txt .
+
+In Nightly builds we have following configuration:
+* Conda builds, depend on: https://anaconda.org/pytorch-nightly/torchtriton
```
Comment (malfet) [inline RELEASE.md:362]: Probably too late to make the change right now, but why not rename it to `pytorch-triton` or just `triton` to be consistent with PyPI?

**Review by malfet — APPROVED** [review]

File: RELEASE.md (hunk 3 above)
Comment (atalman) [reply RELEASE.md:362]: Agree with this for release 2.2 we should probably do it. To make it less confusing for people

File: RELEASE.md (hunk 2 above)
Comment (atalman) [reply RELEASE.md:371]: The release 2.x branch does not exist, I want to make sure it points to the existing branch.

File: RELEASE.md (hunk 1 above)
Comment (atalman) [reply RELEASE.md:369]: No it's not this is the correct link. AMD confirmed

**Pull request closed without merging**